### PR TITLE
Add new leave type UNPAID:

### PR DESCRIPTION
**Pull request by RohitNalePUCSD**
- 1. Add UNPAID leave type in global constant.
 - 2. Add UNPAID, WFH, SPL leave description on apply leave page.
 - 3. Add condition when user avl leave is 0 then default show UNPAID leave option.
 - 4. Added test cases of above all.

### CODE REVIEW

**Comment by RohitNalePUCSD** [comment]
[Add new leave type as “UNPAID”](https://trello.com/c/tUBBJnag/163-add-new-leave-type-as-unpaid)

File: app/views/leave_applications/_form.html.haml [hunk 1]
```diff
@@ -1,13 +1,23 @@
...
+    = f.input :number_of_days, as: :decimal, label: 'Days', input_html: {step: 1}, input_html: { required: true, :'data-validation-required-message' => 'Enter the number of days', min: "1", :'data-validation-min-message' => "Number of day can't be zero or negative"}, readonly: true
+    = f.input :contact_number, as: :decimal, label: 'Emergency Contact No.', input_html: {required: true, :'data-validation-required-message' => 'You need to enter your emergency number', type: 'text', maxlength: 10}
+    = f.input :reason, as: :text, label: 'Reason', input_html: {required: true, :'data-validation-required-message' => 'You have to specify valid reason for leave'}
+    - @end_of_year = Date.today.end_of_year.strftime('%Y-%m-%d')
+    - @location = current_user.country
+  .span3
+    %center
+      %p 1. WFH: To mention "Work From Home" availability, here available leave count remains unchanged.<br><br>2. SPL: Special leaves such as Paternity or Maternity leaves, Comp off, etc. Same as above, the available leave count remains unchanged.<br><br>3. UNPAID: If you want to take a leave and available leaves are 0 then kindly select Unpaid.
```
Comment (Sameer-Tilak) [inline app/views/leave_applications/_form.html.haml:18]: Change this statement as 

1. WFH: Work From Home"
2. SPL: Special leaves such as Paternity or Maternity leaves, Comp off, etc. 
3. UNPAID: Apply this when available leave balance is 0 (Zero)

**Comment by Sameer-Tilak** [comment]
Comment added. Please consider.

File: app/views/leave_applications/_form.html.haml (hunk 1 above)
Comment (RohitNalePUCSD) [reply app/views/leave_applications/_form.html.haml]: Done!!

**Pull request merged**